src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +110,52 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestRoeUpdate(HttpServletRequest request) {
```

Comment:
To avoid storing company number as a class level variable, probably easiest to define a new local, private 'pair' POJO that contains the boolean result and company number and return it from this method.

Reply:
Reworked it it in another way !